### DISCUSSION TITLE
Brax v2 preview is out!

### DISCUSSION THREAD

**Discussion by btaba** [google/brax#286] [Announcements]
Hi folks, we have released a Brax v2 preview [here](https://github.com/google/brax/tree/main/brax/v2).

Some highlights from the release notes:
* Generalized physics backend.
* Continued support for the Spring physics backends. PBD will soon follow.
* Direct support for Mujoco XML format, and URDF by association.
* Fully traceable System object.
* Env API that better supports custom physics backends.
* Open sourced visualizer server.

Be sure to check out the [training notebook](https://colab.sandbox.google.com/github/google/brax/blob/main/notebooks/Brax_v2_Training_Preview.ipynb) as well.

For folks who want to pin v1 brax, you can use [v0.0.16](https://github.com/google/brax/releases/tag/v0.0.16). Stay tuned for a major release of brax v2.

**Comment by ManuCorrea** [comment]
Hello @btaba,

I would like to contribute to the project, is there any issue/item that can be addressed from the community? I also saw the only issue with good first issue is this one https://github.com/google/brax/issues/66, does this still relevant? Could any other issues be suitable as good first one?

**Reply by btaba** [reply]
Hi @ManuCorrea ! We haven't been great about managing tags, we'll try to be a bit better going forward. Here are a couple of suggestions of things you could look into:
* https://github.com/google/brax/issues/224
* loading textures from mujoco and piping them to the visualizer, pointers:  [1](https://github.com/google/brax/blob/main/brax/v2/io/mjcf.py#L293), [2](https://mujoco.readthedocs.io/en/stable/XMLreference.html#asset-material)
* sphere or capsule <> heightmap collisions

@cdfreeman-google @erikfrey any other ideas?

**Reply by ManuCorrea** [reply]
@btaba I have been trying to enhance the JS GUI but it hasn't been possible to me.

viewer.js gets imported from https://cdn.jsdelivr.net/gh/google/brax@v0.1.1/brax/v2/visualizer/js/viewer.js so local changes don't apply.

I have been trying several ways of importing it, relatives imports from index.html, hardcoded path in index and in the render (html.py) functions but I haven't found the way.

Which is the proper way to develop for the front-end part locally?

Sorry if there is some obvious way to it but I have spend quite some try just trying to get it running instead of coding the feature.

Thank you in advance.

**Reply by btaba** [reply]
Hi @ManuCorrea ! You can run the visualizer server `python3 brax/v2/visualizer/visualizer.py` to develop locally! We have a couple of missing deps in setup.py that we'll have added

**Reply by ManuCorrea** [reply]
It was kinda obvious but didn't notice it existed! Thank you a lot :) I also added the missing deps on setup, assuming this is a main tool not just a develop tool

**Comment by moskomule** [comment]
Hi @btaba! When do you plan to make the next major release (v2) public? I would like to know the (approximated) timeline.

**Reply by btaba** [reply]
Hi @moskomule ! We are implementing the pbd backend and all the same environments as in v1 (and resolving issues thereof). It's hard to pin down an exact date but we are on the tail end of getting v2 out

**Reply by Theo-Cheynel** [reply]
@moskomule v2 is out apparently :tada:

**Comment by bayerj** [comment]
Hey,

I was looking through the code to see if I could build my own simple environment to play with the traceable system state. Currently, only the Ant env seems to have been ported, which is rather complicated. It would be really nice to have something radically simple, for example the reacher, to play with that.

Thanks for the great work!

**Reply by btaba** [reply]
Hi @bayerj we just pushed several more envs, including `reacher`!

**Reply by kallel-mahdi** [reply]
Hi @btaba do you have an (approximate) idea when the Swimmer environment will be available in the v2 release? Many thanks!

**Reply by btaba** [reply]
Hi @kallel-mahdi , I don't think we'll be adding swimmer, but if you'd like to add that please feel free! We'll be moving v2 to the mainline very shortly O(days)

**Comment by btaba** [comment]
Quick update, we've moved brax v2 to the main folder in v0.9.0. `brax.v1` is available if folks still want to use it but it's being deprecated. We will bump to a major 1.0.0 release once we hash out a few last bits!

**Comment by yardenas** [comment]
Hi!
I'd also like to contribute, any “good-first-issues” I can start working on?

Cheers :rocket:
